lib/state_server/state.ex

```diff
@@ -82,6 +82,16 @@ defmodule StateServer.State do
   defer handle_call
   ```
 
+  #### Important
+
+  If you handle an event via **any** instance of a handler function block in
+  the main `StateServer` module, and return a  `:reply` or `:noreply`, it
+  will **not** be handled by the `State` module, you must explicitly
+  specify `:defer` to be handled by `State` modules.
```

Comment:
this note doesn't cover the case I'm looking for (though it is helpful):

namely, what if I have both `handle_transition/3` and `on_state_entry/3` in the main `StateServer` module. I expect any update events from `handle_transition/3` to appear in `on_state_entry/3`, but it seems only one of them (the last one?) persists to `data`.